Feat: detailed plugin list

FMotalleb opened the pull request:
A workflow to keep track of nu-plugin and nu-protocol version of the plugins

fdncred commented:
I think this really looks super! Thanks! The only things that kind of concerns me of this is...
1. When we add new plugins, I think we need to add it in this CI. It would be great to have a short readme that says, "When you add new plugins, do this..."
2. Hopefully the syntax of nushell isn't going to change greatly, but if it does, that's a pretty big script. Not complaining really, just not wanting to change it if we have some breaking syntax change.
3. Lastly, I think we need a link off of the main readme.md that points to the detailed list. I still want to maintain the one-liner plugins section, but we should link to the details somewhere.

FMotalleb commented:
I understand, but the workflow asks what version of nushell it uses so I think simple nushell updates (solely) won't break this workflow but upgrading the workflow's nushell version may cause this issue.

added a link to `plugin_details.md` under the plugin section (it is broken for now because it links it to the `nushell/awesome-nu` and the file is not there yet)

added a link and an instruction on how to add your own plugin under the plugin section.

fdncred reviewed (approved):
looks good now. thanks for all this!
